src/Sylius/Bundle/AddressingBundle/spec/Form/EventListener/BuildAddressFormSubscriberSpec.php

```diff
@@ -34,12 +36,12 @@ function let(ObjectRepository $countryRepository, FormFactoryInterface $factory)
 
     function it_is_initializable()
     {
-        $this->shouldHaveType('Sylius\Bundle\AddressingBundle\Form\EventListener\BuildAddressFormSubscriber');
+        $this->shouldHaveType(BuildAddressFormSubscriber::class);
```

Comment:
@pamil how you want to get FQCN of class which does not exist? 
Writing this:
```php
MyClass::class
```

Will not return the full namespace, because the class does not exist and the namespace is unknow. IMO `shouldHaveType` should always be string.

Regarding `implements` we can discuss it separately.